cmd/gcp-controller-manager/node_syncer.go

```diff
@@ -156,13 +191,36 @@ func (ns *nodeSyncer) process(key string) error {
 	if err != nil {
 		return fmt.Errorf("failed to get Pod %q: %v", key, err)
 	}
-	if !exists {
-		node, gsa, found := ns.nodes.remove(key)
-		if !found {
+	if !exists { // pod removal event
+		podUIDs := ns.nodes.find(key)
```

Comment:
Given we have to account for pods with duplicate names, should we be keying this sync loop off of pod UID instead of metanamespacekeyfunc? Does that avoid the need for this expensive find and the need to potentially sync multiple nodes on a single pod update?

Reply:
Are you referring to the podMap that this control loop is operating upon?  This code is now keying pods by their UID to handle duplicate names; see l265.  The expensive find is to find all the UIDs (normally singular) by the namespace/name which is all we get from the event.

Did I understand you comment correctly?